codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeSpecUtils.java

```diff
@@ -175,6 +170,27 @@ public Map<List<String>, List<AuthType>> operationsToAuthType() {
         return operationsToAuthType;
     }
 
+    public List<AuthType> serviceDefaultAuthTypes() {
+        List<AuthType> modeled = intermediateModel.getMetadata().getAuth();
+        if (!modeled.isEmpty()) {
+            return modeled;
+        }
+        return Collections.singletonList(intermediateModel.getMetadata().getAuthType());
+    }
+
+    public List<AuthType> allServiceAuthTypes() {
+        Set<AuthType> result =
+            intermediateModel.getOperations()
+                             .values()
+                             .stream()
+                             .map(OperationModel::getAuth)
+                             .flatMap(List::stream)
+                             .collect(Collectors.toSet());
+        List<AuthType> modeled = intermediateModel.getMetadata().getAuth();
+        result.addAll(modeled);
+        return result.stream().sorted().collect(Collectors.toList());
```

Comment:
why are you sorting here?

Reply:
We use the output of this method to generate the map of scheme id to scheme class. If we don't sort the output of such method will be non-deterministic and prone to change. E.g.,

```
    private Map<String, AuthScheme<?>> defaultAuthSchemes() {
        AwsV4AuthScheme awsV4AuthScheme = AwsV4AuthScheme.create();
        BearerAuthScheme bearerAuthScheme = BearerAuthScheme.create();
        return MapUtils.of(awsV4AuthScheme.schemeId(), awsV4AuthScheme, bearerAuthScheme.schemeId(), bearerAuthScheme);
    }
```